rules.neon

```diff
@@ -19,6 +19,7 @@ rules:
 	- PHPStan\Rules\Doctrine\ORM\DqlRule
 	- PHPStan\Rules\Doctrine\ORM\RepositoryMethodCallRule
 	- PHPStan\Rules\Doctrine\ORM\EntityRelationRule
+	- PHPStan\Rules\Doctrine\ORM\EntityNotFinalRule
```

Comment:
It has to be added to `services`. This way you're always registering the rule and the `conditionalTags` section has no effect.